packages/devtools-evm-hardhat/src/tasks/simulation/start.ts

```diff
@@ -32,12 +34,24 @@ const action: ActionType<SimulationStartTaskArgs> = async (
 
     const logger = createLogger()
 
+    // --stage cannot be used in conjunction with --networks
+    if (networksArgument != null && stage != null) {
+        logger.error(`--stage ${stage} cannot be used in conjunction with --networks ${networksArgument.join(',')}`)
+
+        process.exit(1)
+    }
+
+    // And we create a filtering predicate for the stage argument
+    const isOnStage = stage == null ? () => true : (eid: EndpointId) => endpointIdToStage(eid) === stage
+
     // Let's grab the networks that will be included in the simulation
     const networks = networksArgument
         ? // Here we need to check whether the networks have been defined in hardhat config
           assertDefinedNetworks(networksArgument)
         : //  But here we are taking them from hardhat config so no assertion is necessary
-          Object.entries(getEidsByNetworkName(hre)).flatMap(([networkName, eid]) => (eid == null ? [] : [networkName]))
+          Object.entries(getEidsByNetworkName()).flatMap(([networkName, eid]) =>
+              eid != null && isOnStage(eid) ? [networkName] : []
```

Comment:
would this be nicer if we used filter()?

Reply:
You would need `filter` and `map` with an additional type declaration, i.e. `flatMap`:

```typescript
Object.entries(getEidsByNetworkName())
  .filter((value): value is [string, EndpointId] => eid != null && isOnStage(eid))
  .map(([networkName]) => networkName)
```